config/config.go

```diff
@@ -290,14 +291,35 @@ func determineTreeRoot(v *viper.Viper, cfg *Config, logger *log.Logger) error {
 		return errors.New("at most one of tree-root, tree-root-cmd or tree-root-file can be specified")
 	}
 
-	// set git-based tree root command if the walker is git and no tree root has been specified
-	if cfg.Walk == walk.Git.String() && count == 0 {
-		cfg.TreeRootCmd = "git rev-parse --show-toplevel"
+	// only attempt the following if no tree root has been provided
+	if count == 0 {
+		setGitRootCmd := func() {
+			cfg.TreeRootCmd = git.TreeRootCmd
 
-		logger.Infof(
-			"git walker enabled and tree root has not been specified: defaulting tree-root-cmd to '%s'",
-			cfg.TreeRootCmd,
-		)
+			logger.Infof(
+				"git repo detected and tree root has not been specified: defaulting tree-root-cmd to '%s'",
+				cfg.TreeRootCmd,
+			)
+		}
+
+		switch {
+		case cfg.Walk == walk.Auto.String():
+			// check if the directory containing the config file is a git repo
+			isGitRepo, err := git.IsInsideWorktree(cfg.WorkingDirectory)
+			if err != nil {
+				return fmt.Errorf("failed to check if working directory is a git repo: %w", err)
+			}
+
+			// if it is, set the default git tree root command
+			if isGitRepo {
+				setGitRootCmd()
+			}
+
+		case cfg.Walk == walk.Git.String():
+			setGitRootCmd()
+
+		default: // otherwise, do nothing
```

Comment:
It didn't before, but should this log anything?

E.g. "No 'tree-root', 'tree-root-file', or 'tree-root-cmd' specified and not in a git repo. Assuming working directory is the tree root: %s"